src/api/account.ts

```diff
@@ -1,35 +1,53 @@
+import { JSONStream } from '../utils.ts'
+
 export class Account {
   constructor(private readonly fetcher: Function) {}
 
-  public profile() {
-    return this.fetcher(`/account`)
+  public async profile() {
+    const response = await this.fetcher(`/api/account`)
+    const json = await response.json()
```

Comment:
dedupe same jobs by letting fetcher handle data

Reply:
I will take a look at ndjson. I agree, the code duplication is a problem.

This will be my plan of implementation for the fetcher:
- Check if endpoint streams ndjson
  - If it does, pipe the stream through `ndjson`
  - Since the stream might be ongoing (e.g. moves of a game), we can't wait for it to end before returning the json.
  - We need to give the (processed) stream to the programmer so they can decide what to do with it (e.g. use `for await` loop to iterate through new chunks)
- If it is not a stream, just parse it as JSON and return it.

Reply:
This way, the fetcher takes care of the processing and parsing and we can just do `return this.fetcher( ... )` as I'm sure you would have wanted :)

Reply:
Yeah, sounds good to me!